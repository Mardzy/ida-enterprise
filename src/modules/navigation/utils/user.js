/**
 * Possible username keys in user object.
 * @type {string[]}
 */
const MAY_BE_USERNAME_KEY = ["username", "userName"];

/**
 * Look up username based in the given user object.
 * @param user
 * @return {*}
 */
export const getUsername = user => MAY_BE_USERNAME_KEY.reduce((result, key) => result || user.get(key), null);

/**
 * Construct a display name for the user.
 * @return {string}
 */
export const getDisplayName = user => {
  /**
   * @type {string}
   */
  const firstName = user.get("givenName");

  /**
   * @type {string}
   */
  const lastName = user.get("surName");

  return [firstName, " ", lastName].join("").trim() || getUsername(user);
};