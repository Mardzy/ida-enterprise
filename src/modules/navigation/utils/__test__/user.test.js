import { fromJS } from "immutable";

import { getUsername, getDisplayName } from "../user";

describe("The user util function", () => {

  it("getUsername retrieves a username from the user object", () => {
    const username = getUsername(fromJS({
      username: "sandor@connecterra.io"
    }));

    return expect(username).toMatchSnapshot();
  });

  it("getDisplayName retrieves a display name from the user object", () => {
    const username = getDisplayName(fromJS({
      givenName: "Sandor",
      surName: "Kocsis"
    }));

    return expect(username).toMatchSnapshot();
  });

});