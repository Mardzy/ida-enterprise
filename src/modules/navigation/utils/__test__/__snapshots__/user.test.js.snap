// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The user util function getDisplayName retrieves a display name from the user object 1`] = `"Sandor Kocsis"`;

exports[`The user util function getUsername retrieves a username from the user object 1`] = `"sandor@connecterra.io"`;
